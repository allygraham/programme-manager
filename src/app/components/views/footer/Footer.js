import React from 'react';
import styled from 'react-emotion';

import { constants } from '~/constants';

const FooterBlock = styled('footer')`
  padding: 1.5rem;
  background-color: #757575;
  color: #FFF;
`;
  
const Link = styled('a')`
  color: #BBDEFB;
  font-weight: 700;
  transition: color 300ms ease-in-out;

  &:hover {
    color: #6BB7F6;
  }
`;

export const Footer = () => (
  <FooterBlock className="footer">
    <div className="content has-text-centered">
      <p>
        Built by <Link href={`mailto:${constants.footer.email}`}>{constants.footer.name}</Link>
      </p>
    </div>
  </FooterBlock>
);
