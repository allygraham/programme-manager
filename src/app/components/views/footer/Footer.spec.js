import React from 'react';
import { shallow } from 'enzyme';

import { Footer } from './Footer';

describe('Footer', () => {

  test('render the <Footer> component', () => {
    const wrapper = shallow(<Footer/>);
    expect(wrapper).toMatchSnapshot();
  });

});
