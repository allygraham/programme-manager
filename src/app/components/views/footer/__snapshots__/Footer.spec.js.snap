// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render the <Footer> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="content has-text-centered"
      >
        <p>
          Built by 
          <Styled(a)
            href="mailto:ally@allygraham.co.uk"
          >
            Ally Graham
          </Styled(a)>
        </p>
      </div>,
      "className": "footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Built by 
          <Styled(a)
            href="mailto:ally@allygraham.co.uk"
          >
            Ally Graham
          </Styled(a)>
        </p>,
        "className": "content has-text-centered",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Built by ",
            <Styled(a)
              href="mailto:ally@allygraham.co.uk"
            >
              Ally Graham
            </Styled(a)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Built by ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Ally Graham",
              "href": "mailto:ally@allygraham.co.uk",
            },
            "ref": null,
            "rendered": "Ally Graham",
            "type": [Function],
          },
        ],
        "type": "p",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="content has-text-centered"
        >
          <p>
            Built by 
            <Styled(a)
              href="mailto:ally@allygraham.co.uk"
            >
              Ally Graham
            </Styled(a)>
          </p>
        </div>,
        "className": "footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Built by 
            <Styled(a)
              href="mailto:ally@allygraham.co.uk"
            >
              Ally Graham
            </Styled(a)>
          </p>,
          "className": "content has-text-centered",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Built by ",
              <Styled(a)
                href="mailto:ally@allygraham.co.uk"
              >
                Ally Graham
              </Styled(a)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Built by ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Ally Graham",
                "href": "mailto:ally@allygraham.co.uk",
              },
              "ref": null,
              "rendered": "Ally Graham",
              "type": [Function],
            },
          ],
          "type": "p",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
